shaders/MicrobeBackground.shader

```diff
@@ -15,6 +18,47 @@ const vec2 speed3 = vec2(1050.0f);
 varying vec2 UV3;
 varying vec2 UV4;
 
+vec3 LightInfluence(float amount)
+{
+    vec3 influence = vec3(amount, amount, amount);
+    int time = 0;
+
+    if(amount > 0.5f)
+        time = 2;
+    else if(amount > 0.25f)
+        time = 1;
+    else
+        time = 0;
```

Comment:
The never use If statements in shaders is really old advice. The software can handle If statements now with almost no performance effect. Also, there is no way an if statement that just sets a value won't immediately get optimized by the compiler.

Reply:
There's an if + a switch right after. Any branching operation in GPU still needs to take all of the cores in an execution group separately through all of the branches, though in this case as the value is an uniform it's not as bad as it could be. But to me the if followed by a switch statement looks just really not optimal graphics programming and triggers alarms in my head.

Reply:
I believe I've addressed this but I'll leave this comment unresolved as it needs checking from someone who understands the requirements of shader code.

Reply:
Unless someone can math formula this out, then I *think* having just one sequence of ifs is *probably* fine enough. I think that maybe a better approach would be to pass the entire multiplier vec as an uniform. That way the c# code can do the calculation once (which should be very cheap with a couple of ifs) and all of the shader units don't need to take conditional branches then.